CMakeLists.txt

```diff
@@ -6,6 +6,9 @@ set(ALEVERSION "0.5")
 option(USE_SDL "Use SDL" OFF)
 option(USE_RLGLUE "Use RL-Glue" OFF)
 option(BUILD_EXAMPLES "Build Example Agents" ON)
+option(BUILD_CPP_LIB "Build C++ Shared Library" ON)
```

Comment:
Setting BUILD_CPP_LIB off but leaving BUILD_EXAMPLES on will fail. Can you add some code that errors in this case?